cmd/group-auto-updater/main.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config/org"
+
+	userV1 "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type options struct {
+	group           string
+	peribolosConfig string
+	org             string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.StringVar(&o.group, "group", "", "The group that will be updated in the cluster.")
+	fs.StringVar(&o.peribolosConfig, "peribolos-config", "", "Peribolos configuration file")
+	fs.StringVar(&o.org, "org", "", "Org from peribolos configuration")
+
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	if len(o.group) == 0 {
+		return fmt.Errorf("--group is not specified")
+	}
+	if len(o.peribolosConfig) == 0 {
+		return fmt.Errorf("--peribolos-config is not specified")
+	}
+	if len(o.org) == 0 {
+		return fmt.Errorf("--org is not specified")
+	}
+
+	return nil
+}
+
+func getUserV1Client() (*userV1.UserV1Client, error) {
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not load cluster clusterConfig: %v", err)
+	}
+
+	userV1Client, err := userV1.NewForConfig(clusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not create user openshift client: %v", err)
+	}
+
+	return userV1Client, nil
+}
+
+func getUsersFromPeribolosConfig(pConfig, pOrg string) ([]string, error) {
+	users := sets.NewString()
+
+	b, err := ioutil.ReadFile(pConfig)
+	if err != nil {
+		return nil, fmt.Errorf("could not read peribolos configuration file: %v", err)
+	}
+
+	var peribolosConfig org.FullConfig
+	if err := yaml.Unmarshal(b, &peribolosConfig); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal peribolos config: %v", err)
+	}
+
+	users.Insert(peribolosConfig.Orgs[pOrg].Admins...)
+	users.Insert(peribolosConfig.Orgs[pOrg].Members...)
+
+	return users.List(), nil
+}
+
+func updateGroup(group, peribolosConfig, org string, groupV1Interface userV1.GroupInterface) error {
+	g, err := groupV1Interface.Get(group, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("couldn't get group from cluster: %v", err)
+	}
+
+	users, err := getUsersFromPeribolosConfig(peribolosConfig, org)
+	if err != nil {
+		return fmt.Errorf("couldn't get users from peribolos config: %v", err)
+	}
+	g.Users = users
+
+	if _, err := groupV1Interface.Update(g); err != nil {
+		return fmt.Errorf("couldn't update group in the cluster: %v", err)
+	}
+
+	return nil
+}
+
+func main() {
```

Comment:
I feel like we are missing a paragraph somewhere that would document what this tool is supposed to do and why it exists.

Reply:
I'll add a readme file.

Reply:
I don't think a full README is actually necessary, all I would like was like two or three sentences in a GoDoc.